src/steps/trackTransaction.ts

```diff
@@ -139,13 +139,20 @@ export function trackTransaction(
       step adds content before the first step, the plugin will attempt to insert tracked replacement for the first change at a position
       that corresponds to the first change position if the second change (second in time but occuring earlier in doc) never occured.
       */
+
+      //when having multiple delete steps with and without dataTracked as insert
+      // deleteNode in newTr will affect next step in the old transaction, so will map next steps to the newTr
+      if (newTr.steps.length && isDelete) {
+        thisStepMapping = newTr.mapping.slice(newTr.steps.length - 1)
+      }
+
       // @TODO - check if needed to be done for other types of steps
       const newStep = new ReplaceStep(
         thisStepMapping.map(invertedStep.from),
         thisStepMapping.map(invertedStep.to),
         invertedStep.slice
       )
-      const stepResult = newTr.maybeStep(isDelete ? invertedStep : newStep)
```

Comment:
Do you think we should produce some tests for these cases? It fills like we change the same thing back and forth and it may return us to some previously fixed bugs?

Reply:
good point. I added test cases from LEAN-3136 & LEAN-3898 to cover that area of code. 

I think we are going to return back in the future to that code block as we still having issues with that case of insertion parent and child node in different `tr.insert` call